### PR TITLE
Adding Timing vs BX plots, and modifying thresholds for laser quality

### PR DESCRIPTION
**Pull request by tanmaymudholkar**
Companion to Deployment PR#455: https://github.com/dmwm/deployment/pull/455

Link to 900 PR: https://github.com/cms-sw/cmssw/pull/18069

- Added new plots for the average partition timing v/s BX number.
- Changed expected values of laser timing from 4.2 clocks to 3.7 clocks for Laser 2. Also changed (the value of the deviation from the expected timing above which bad quality is triggered) to 1.0.

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
please test

**Comment by dmitrijus** [comment]
+1

File: DQM/EcalMonitorTasks/interface/TimingTask.h [hunk 1]
```diff
@@ -19,10 +19,18 @@ namespace ecaldqm {
...
+    enum Constants {
+      nBXBins = 15
+    };
+
   private:
     void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+    void beginEvent(edm::Event const&, edm::EventSetup const&) override;
     void setParams(edm::ParameterSet const&) override;
 
+    int bxBinEdges_[nBXBins + 1];
```
Comment (davidlange6) [inline DQM/EcalMonitorTasks/interface/TimingTask.h]: use std::array<int,nBXBins+1>
@tanmaymudholkar

Comment (tanmaymudholkar) [inline DQM/EcalMonitorTasks/interface/TimingTask.h]: Hi @davidlange6 , thanks for the suggestion, but using std::array is inconsistent with the syntax used here: https://github.com/tanmaymudholkar/cmssw/blob/TimingVsBX_and_LaserTiming_91X/DQM/EcalMonitorTasks/src/TimingTask.cc#L65

I used this construction to remain consistent with another very similar plot that we've had for some time: https://github.com/tanmaymudholkar/cmssw/blob/TimingVsBX_and_LaserTiming_91X/DQM/EcalMonitorTasks/src/TrigPrimTask.cc#L79-L80

Perhaps I could try to replace both of them by std::vectors that read in the bin values from the config file? Do you know if that would work with std::upper_bound()? That might take me some time (~1 day) to test etc. Please advise.

Comment (davidlange6) [inline DQM/EcalMonitorTasks/interface/TimingTask.h]: did this not work?

int* pBin(std::upper_bound(bxBinEdges_.begin(), bxBinEdges_.end(), _evt.bunchCrossing()));

Comment (tanmaymudholkar) [inline DQM/EcalMonitorTasks/interface/TimingTask.h]: Hi @davidlange6 , thanks very much for that suggestion. It does indeed improve code readability. I've tested the changes and added a commit to this PR and to the 900 PR: https://github.com/cms-sw/cmssw/pull/18069

Please let me know if you have other suggestions. Thanks!

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**